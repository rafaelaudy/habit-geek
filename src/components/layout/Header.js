import React from "react";
import { Link } from "@reach/router";
import { useTranslation } from "react-i18next";

import "./Header.scss";

function Header({ jwt }) {
  const { t } = useTranslation();

  const getLinkClass = ({ isCurrent }) => {
    return {
      className: isCurrent ? "nav-item nav-link active" : "nav-item nav-link"
    };
  };

  let userLink, userLabel;
  if (jwt) {
    userLink = "/habit-geek/profile";
    userLabel = t("nav-profile");
  } else {
    userLink = "/habit-geek/login";
    userLabel = t("nav-login");
  }

  return (
    <header>
      <nav className="navbar navbar-expand navbar-dark bg-dark header">
        <Link className="navbar-brand" to="/habit-geek/habits">
          Habit Geek!
        </Link>
        <div className="navbar-nav">
          <Link getProps={getLinkClass} to="/habit-geek/habits">
            {t("nav-my-habits")}
          </Link>
          <Link getProps={getLinkClass} to="/habit-geek/history">
            {t("nav-history")}
          </Link>
          <Link getProps={getLinkClass} to={userLink}>
            {userLabel}
          </Link>
        </div>
      </nav>
    </header>
  );
}

export default Header;
