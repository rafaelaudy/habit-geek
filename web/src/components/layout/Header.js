import React from "react";
import { Link } from "@reach/router";
import { useTranslation } from "react-i18next";

import "./Header.scss";

function Header(props) {
  const { t } = useTranslation();

  const getLinkClass = ({ isCurrent }) => {
    return {
      className: isCurrent ? "nav-item nav-link active" : "nav-item nav-link"
    };
  };

  return (
    <header>
      <nav className="navbar navbar-expand navbar-dark bg-dark header">
        <Link className="navbar-brand" to="/">
          Habit Geek!
        </Link>
        <div className="navbar-nav">
          <Link getProps={getLinkClass} to="/habits">
            {t("nav-my-habits")}
          </Link>
          <Link getProps={getLinkClass} to="/history">
            {t("nav-history")}
          </Link>
          <Link getProps={getLinkClass} to="/profile">
            {t("nav-profile")}
          </Link>
        </div>
      </nav>
    </header>
  );
}

export default Header;
