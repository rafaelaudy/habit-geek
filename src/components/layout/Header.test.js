import React from "react";
import { mount } from "enzyme";
import Header from "./Header";

describe("Header component", () => {
  it("Renders static elements", () => {
    const header = mount(<Header />);
    expect(header).toMatchSnapshot();
  });

  it("Changes login to profile when jwt is present", () => {
    const header = mount(<Header jwt={"jwt"} />);
    expect(header).toMatchSnapshot();
  });
});
