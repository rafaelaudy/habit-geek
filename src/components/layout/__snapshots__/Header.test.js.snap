// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Changes login to profile when jwt is present 1`] = `
<Header
  jwt="jwt"
>
  <header>
    <nav
      className="navbar navbar-expand navbar-dark bg-dark header"
    >
      <ForwardRef
        className="navbar-brand"
        to="/habit-geek/habits"
      >
        <Location>
          <LocationProvider
            history={
              Object {
                "_onTransitionComplete": [Function],
                "listen": [Function],
                "location": Object {
                  "assign": [Function],
                  "hash": "",
                  "host": "localhost",
                  "hostname": "localhost",
                  "href": "http://localhost/",
                  "key": "initial",
                  "origin": "http://localhost",
                  "pathname": "/",
                  "port": "",
                  "protocol": "http:",
                  "reload": [Function],
                  "replace": [Function],
                  "search": "",
                  "state": null,
                  "toString": [Function],
                },
                "navigate": [Function],
                "transitioning": false,
              }
            }
          >
            <a
              className="navbar-brand"
              href="/habit-geek/habits"
              onClick={[Function]}
            >
              Habit Geek!
            </a>
          </LocationProvider>
        </Location>
      </ForwardRef>
      <div
        className="navbar-nav"
      >
        <ForwardRef
          getProps={[Function]}
          to="/habit-geek/habits"
        >
          <Location>
            <LocationProvider
              history={
                Object {
                  "_onTransitionComplete": [Function],
                  "listen": [Function],
                  "location": Object {
                    "assign": [Function],
                    "hash": "",
                    "host": "localhost",
                    "hostname": "localhost",
                    "href": "http://localhost/",
                    "key": "initial",
                    "origin": "http://localhost",
                    "pathname": "/",
                    "port": "",
                    "protocol": "http:",
                    "reload": [Function],
                    "replace": [Function],
                    "search": "",
                    "state": null,
                    "toString": [Function],
                  },
                  "navigate": [Function],
                  "transitioning": false,
                }
              }
            >
              <a
                className="nav-item nav-link"
                href="/habit-geek/habits"
                onClick={[Function]}
              >
                nav-my-habits
              </a>
            </LocationProvider>
          </Location>
        </ForwardRef>
        <ForwardRef
          getProps={[Function]}
          to="/habit-geek/history"
        >
          <Location>
            <LocationProvider
              history={
                Object {
                  "_onTransitionComplete": [Function],
                  "listen": [Function],
                  "location": Object {
                    "assign": [Function],
                    "hash": "",
                    "host": "localhost",
                    "hostname": "localhost",
                    "href": "http://localhost/",
                    "key": "initial",
                    "origin": "http://localhost",
                    "pathname": "/",
                    "port": "",
                    "protocol": "http:",
                    "reload": [Function],
                    "replace": [Function],
                    "search": "",
                    "state": null,
                    "toString": [Function],
                  },
                  "navigate": [Function],
                  "transitioning": false,
                }
              }
            >
              <a
                className="nav-item nav-link"
                href="/habit-geek/history"
                onClick={[Function]}
              >
                nav-history
              </a>
            </LocationProvider>
          </Location>
        </ForwardRef>
        <ForwardRef
          getProps={[Function]}
          to="/habit-geek/profile"
        >
          <Location>
            <LocationProvider
              history={
                Object {
                  "_onTransitionComplete": [Function],
                  "listen": [Function],
                  "location": Object {
                    "assign": [Function],
                    "hash": "",
                    "host": "localhost",
                    "hostname": "localhost",
                    "href": "http://localhost/",
                    "key": "initial",
                    "origin": "http://localhost",
                    "pathname": "/",
                    "port": "",
                    "protocol": "http:",
                    "reload": [Function],
                    "replace": [Function],
                    "search": "",
                    "state": null,
                    "toString": [Function],
                  },
                  "navigate": [Function],
                  "transitioning": false,
                }
              }
            >
              <a
                className="nav-item nav-link"
                href="/habit-geek/profile"
                onClick={[Function]}
              >
                nav-profile
              </a>
            </LocationProvider>
          </Location>
        </ForwardRef>
      </div>
    </nav>
  </header>
</Header>
`;

exports[`Header component Renders static elements 1`] = `
<Header>
  <header>
    <nav
      className="navbar navbar-expand navbar-dark bg-dark header"
    >
      <ForwardRef
        className="navbar-brand"
        to="/habit-geek/habits"
      >
        <Location>
          <LocationProvider
            history={
              Object {
                "_onTransitionComplete": [Function],
                "listen": [Function],
                "location": Object {
                  "assign": [Function],
                  "hash": "",
                  "host": "localhost",
                  "hostname": "localhost",
                  "href": "http://localhost/",
                  "key": "initial",
                  "origin": "http://localhost",
                  "pathname": "/",
                  "port": "",
                  "protocol": "http:",
                  "reload": [Function],
                  "replace": [Function],
                  "search": "",
                  "state": null,
                  "toString": [Function],
                },
                "navigate": [Function],
                "transitioning": false,
              }
            }
          >
            <a
              className="navbar-brand"
              href="/habit-geek/habits"
              onClick={[Function]}
            >
              Habit Geek!
            </a>
          </LocationProvider>
        </Location>
      </ForwardRef>
      <div
        className="navbar-nav"
      >
        <ForwardRef
          getProps={[Function]}
          to="/habit-geek/habits"
        >
          <Location>
            <LocationProvider
              history={
                Object {
                  "_onTransitionComplete": [Function],
                  "listen": [Function],
                  "location": Object {
                    "assign": [Function],
                    "hash": "",
                    "host": "localhost",
                    "hostname": "localhost",
                    "href": "http://localhost/",
                    "key": "initial",
                    "origin": "http://localhost",
                    "pathname": "/",
                    "port": "",
                    "protocol": "http:",
                    "reload": [Function],
                    "replace": [Function],
                    "search": "",
                    "state": null,
                    "toString": [Function],
                  },
                  "navigate": [Function],
                  "transitioning": false,
                }
              }
            >
              <a
                className="nav-item nav-link"
                href="/habit-geek/habits"
                onClick={[Function]}
              >
                nav-my-habits
              </a>
            </LocationProvider>
          </Location>
        </ForwardRef>
        <ForwardRef
          getProps={[Function]}
          to="/habit-geek/history"
        >
          <Location>
            <LocationProvider
              history={
                Object {
                  "_onTransitionComplete": [Function],
                  "listen": [Function],
                  "location": Object {
                    "assign": [Function],
                    "hash": "",
                    "host": "localhost",
                    "hostname": "localhost",
                    "href": "http://localhost/",
                    "key": "initial",
                    "origin": "http://localhost",
                    "pathname": "/",
                    "port": "",
                    "protocol": "http:",
                    "reload": [Function],
                    "replace": [Function],
                    "search": "",
                    "state": null,
                    "toString": [Function],
                  },
                  "navigate": [Function],
                  "transitioning": false,
                }
              }
            >
              <a
                className="nav-item nav-link"
                href="/habit-geek/history"
                onClick={[Function]}
              >
                nav-history
              </a>
            </LocationProvider>
          </Location>
        </ForwardRef>
        <ForwardRef
          getProps={[Function]}
          to="/habit-geek/login"
        >
          <Location>
            <LocationProvider
              history={
                Object {
                  "_onTransitionComplete": [Function],
                  "listen": [Function],
                  "location": Object {
                    "assign": [Function],
                    "hash": "",
                    "host": "localhost",
                    "hostname": "localhost",
                    "href": "http://localhost/",
                    "key": "initial",
                    "origin": "http://localhost",
                    "pathname": "/",
                    "port": "",
                    "protocol": "http:",
                    "reload": [Function],
                    "replace": [Function],
                    "search": "",
                    "state": null,
                    "toString": [Function],
                  },
                  "navigate": [Function],
                  "transitioning": false,
                }
              }
            >
              <a
                className="nav-item nav-link"
                href="/habit-geek/login"
                onClick={[Function]}
              >
                nav-login
              </a>
            </LocationProvider>
          </Location>
        </ForwardRef>
      </div>
    </nav>
  </header>
</Header>
`;
